### DISCUSSION TITLE
Welcome to UnlockEdv2 Discussions!

### DISCUSSION THREAD

**Discussion by PThorpe92** [UnlockedLabs/UnlockEdv2#169] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about that may relate to something of a broader scope of the project.
  * Share thoughts and ideas about feature requests and about how users interact with UnlockEd.
  * Engage with other community members.
  * Welcome others and are open-minded.

  To get started, comment below with an introduction of yourself and tell us about what interests you about this community.

Hey everyone,

I'm Preston and I'm the lead developer at Unlocked Labs. I'm currently incarcerated at MountainView correctional facility in Charleston, Maine until around May of '25. My full 'story' can be read [here](https://pthorpe92.github.io/intro/my-story) on my blog. 
But the short version is that I am a huge nerd, I love vim, terminals, CLI tools, linux and Rust, and enjoy working on challenging projects.
I hope to continue to work with Unlocked Labs and in the prison reform and education space upon my release, as I'm very passionate about what we are doing and about making prison a place that fosters the kinds of change I've seen in my life through the opportunities I've been fortunate enough to be given.

**Comment by techeclecticdesign** [comment]
Hello everybody,

My name is Chris.  I am currently incarcerated at Maine State Prison and my earliest possible release date is late 2026.  Prior to incarceration I was just beginning to transition from programming as a hobby to programming as a career.  Since I started my time here, I have found opportunities to sharpen my skills, especially since the prison allowed internet access to qualified residents a little over a year ago.

On the outside my life was very hectic, where I struggled to care for a large number of dependents with limited resources.  I have replaced that pressure with a strong work ethic and drive to be successful.  I want to believe that I can have a good career despite the stigma of being a convict, and that dream is starting to come alive more and more by the day.

I am interested in UnlockedLabs and their projects for many reasons.  UnlockEdv2 is in the early stages of development, which is a marvelous time for a project, where creativity shines and there are many options to make helpful contributions.  The community is developing projects which benefit other incarcerated people across the country, which is naturally something I have a lot of support for.

As a whole, the community seems to be a group of people who are either justice affected, or who sympathize with those who are.  I feel that with this community, my status as an incarcerated person, the technological limitations that come with that, and some of the areas where I lack experience, will be more understood.  It seems like a great place for me to build myself up towards my goals, and I'm glad that I discovered this project.

**Comment by calisio** [comment]
Hi everyone, my name is Carolina and I'm one of the project maintainers of UnlockEdv2! I’m a software engineer on the Unlocked Labs team, where I primarily focus on frontend development. 

Most of my previous experience comes from being self-taught or learning on the job (I’m always drawn to startups/early stage projects). Outside of coding, I love playing soccer, reading, and decompressing through art—mainly pottery and b&w film photography. 

I have always been passionate about criminal justice reform and ed tech, and was determined to find a career that intersected the two. I am lucky to have found the Unlocked Labs team early in my career, where I have the opportunity to contribute to a really meaningful project that directly impacts individuals who are incarcerated. Super excited for all the amazing work we'll be doing moving forward!

**Comment by CK-7vn** [comment]
### Hey there! 
My name's Clyde, but everyone calls me Keighan. I'm currently incarcerated at the Bolduc Correctional Facility in Warren, Maine and have hopes to be out on the Maine Department of Corrections home-confinement program by the summer of 2026. I'm incredibly passionate about technology, alternative paths to traditional degrees ("hacking college"), and trying to make the community and lives of people around me better in any way that I can. 

My journey through incarceration has given me a profound understanding of education's transformative power in breaking the cycle of mass incarceration and reducing recidivism. Over the past year, I've earned 86 college credits and an associate's degree through alternative learning pathways, proving that higher education is not only possible but more accessible than many believe, even within prison walls.
UnlockedLabs' mission resonates deeply with my personal experience and advocacy efforts. The teams work to expand educational opportunities for those affected by the justice system aligns perfectly with the message I've been championing at my current facility. The potential for education to change lives and reduce reoffending rates cannot be overstated.
The night UnlockedLabs' mission was explained to me truly felt like a moment of serendipity. The approach of leveraging technology and open-source solutions to bridge the educational gap for incarcerated individuals is innovative and crucial, and I hope to be a help in anyway that I can!

**Comment by jtucholski** [comment]
Hi, my name is Josh. I joined Unlocked Labs about two months ago and have been getting up to speed with the UnlockEd product. This is one of my first experiences maintaining and contributing to an open-source project, as most of my previous work has been closed-source for companies or their clients.

I'm passionate about using technology to positively impact individuals and communities. I've spent a lot of time teaching and helping others find career paths they enjoy, and now I'm excited to apply that same energy to support causes focused on criminal justice reform. I believe it's important to focus on improving the lives of those impacted by the justice system to help them and the communities they all return to.

I live in Cleveland, Ohio. I'm often seen outdoors (we have a National Park) running, biking, or hiking. I also have two young sons who share an interest in tech, so whenever I have the time, I enjoy engaging with them. Whether it's modding Minecraft, building tons of houses and redstone circuits, or exploring coding tutorials together, it's always a good experience for all of us.

-- Josh

**Comment by carddev81** [comment]
Hey everyone, my name is Rich and I was recently blessed to be a part of the UnlockedLabs team. I am super excited to be a part of the team. I am truly living my dream. My current role is software developer apprentice and I have learned so much already. I am a passionate self-taught software developer always reaching for more to learn. 

I have been programming for 13 years (almost) and learned to program while incarcerated at the Jefferson City Correctional Center. I had worked for the Missouri Office of Administration ITSD within the Information Systems Unit at JCCC for over 12 years. I learned how to program by reading book after book and working through many hands-on exercises. It wasn't until a light-bulb moment I had, that I truly grasped the ins and outs of programming. That moment entailed a debug session within an Eclipse IDE when there was a moment where an object reference was being passed into a method and when that reference returned I hovered over the reference variable to open up a popup that displayed the state of the object and noticed that the state within the object had magically changed--that moment of noticing the change is when "I saw the light". From then on I have been eager to learn whatever technology/language is presented to me and to tackle solving any problem.

While I was incarcerated, I had wondered why there were no educational programs available other than the required HiSET, particularly college-level courses. I never thought I'd see a coding program introduced in the prison system, so when I heard that the LaunchCode program was going to begin I was shocked and excited. To all of us it was a miracle! I thought man, this is what I love doing so, of course I signed up and soaked in as much information as I could. It is such an amazing gift for everyone to get the chance be able to learn programming.

I believe the LaunchCode program may have kick-started the Department of Corrections to introduce other educational opportunities, because not long after, they launched the Lincoln University program. I also enrolled into that program and I was accepted. I am so pumped to be a part of the UnlockedLabs team, which will not only ensure that education will thrive within prisons but also will significantly boost the chances for those who are incarcerated to achieve their dreams. This is a much-needed initiative that promises to be highly beneficial.

I almost forgot to say what I do when I'm not writing code. I love fishing, preferably for largemouth bass. I recently missed out on a big catch due to my fishing line snapping--bad moment.  I also enjoy shooting pool, lifting weights, and running. One last thing about myself that I'd like to share is that I learned how to cut my own hair--i save $20 per haircut.

**Comment by Vexdz007** [comment]
I am going to be truthful and not question any ethical concerns of the project.   I tried to make a change using the courts and it didn't go easy or attempt the changes your efforts already achieved by providing a blue print to do it better. I know you are doing the right thing and that there is more to do.  How can I help you.

**Reply by CK-7vn** [reply]
Hi, so nice to meet you! Have you been able to get the product running locally? If you don't mind me asking, what's your experience with the technology stack?

**Reply by Vexdz007** [reply]
I have not even looked at the stack in any of the ide that I have been trying for the projects that I been tinkering with.  I am just a blockchain brat of some experience trying to fix this industry and treatment using contingency management tools for verification and create a better solution to the suffering for subsidize that create more punishment than crime has even attempted.  So sobercoin is now focusing on prison re-entry because of Hippa and other legal prohibited actions to get people's insurance from making patients into a investment.  But prisoners that is not that ethical dilemma and solves many of the same problems that I have already played out using the the time and money they pay to avoid solving the causes that  research has proven work if tried.  I'LL take this down as this is the first time in history that this will work and the town I live in is breaking ground on $150 million dollar new prison and the chance to do  it differently before I buy a 🍌  for bitcoin and call it art in Miami and try and pay the fines for the girl that I care enough to spend the time trying to get a lawyer to walk into court and use her mugshot with a bitcoin stuffed in a NFT to pay for her fines and create a collection for the purchase of the prison for a cold storage and crypto RWA idea that calculated a person's time to pay back society on the actual cost of the lost opportunity punishment for addicts and how much money spent on her books and the phone was costing me as a weeks commissary was half of a million dollars.  You bet I have a moral duty and be honored to use the next few months marketing a good idea for a lifelong potential customer base and novel way to fix a problem that has vexdz me for 5 years now.
